File: src/main/java/com/github/reviversmc/themodindex/api/data/IndexJson.java
package com.github.reviversmc.themodindex.api.data;

import java.util.List;
import java.util.Optional;

/**
 * Represents the index.json file found in the mod index.
 *
 * @param schemaVersion The schema version of the index.json file.
 * @param files         The {@link IndexFile} entries found in the index.json file.
 */
public record IndexJson(Optional<String> schemaVersion, List<IndexFile> files) {
    /**
     * An file entry found in the index.json file.
     * @param identifier The identifier of the mod file, in the format "modLoader:modName:version".
     * @param sha1Hash  The SHA-1 hash of the mod file.
     */
    public record IndexFile(Optional<String> identifier, Optional<String> sha1Hash) {
    }
}

File: src/main/kotlin/com/github/reviversmc/themodindex/api/data/IndexJson.kt
package com.github.reviversmc.themodindex.api.data

/**
 * Represents the index.json file found in the mod index.
 *
 * @param indexVersion The schema version of the index.json file.
 * @param identifiers        The identifier of the mod file, in the format "modLoader:modName:sha512hash".
 * @author ReviversMC
 * @since 5.0.0
 */
@kotlinx.serialization.Serializable
data class IndexJson(val indexVersion: String, val identifiers: List<String>)
